lib/healthcheck/government_data_check.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Healthcheck
+  class GovernmentDataCheck
+    attr_reader :government_repo
+    def initialize
+      @government_repo = BulkData::GovernmentRepository.new
+    end
+
+    def name
+      :government_data_check
+    end
+
+    def status
+      return GovukHealthcheck::CRITICAL unless government_repo.cache_populated?
+      return GovukHealthcheck::WARNING if government_repo.cache_age > 21600
+
+      GovukHealthcheck::OK
+    end
+
+    def details
+      {
+        critical: "No government data availible",
+        warning: warning_details_content,
+      }
```

Comment:
Sorry I missed this before. Is it normal that these methods will return the critical and warning messages even if the healthcheck is not in a critical or warning state? It feels like that is information that'd show in icinga.

Having a look around I'm not too sure what the purpose and structure of details needs to be. It seems common to have a value, total and thresholds (https://github.com/alphagov/govuk_app_config/blob/master/lib/govuk_app_config/govuk_healthcheck/threshold_check.rb#L23-L32) though I don't know what icinga parses or not.

Reply:
I could have sworn that I'd seen it done like this in email alert api but I can't find it now. It does seem to be how [DGU does it](https://github.com/alphagov/datagovuk_publish/blob/a8468e62cb94ff755215b9cf23da345c5f4c31bc/app/models/healthcheck/recurring_jobs/ckan_org_sync.rb#L25) but you're right at this doesn't look entirely normal. I'll have a crack at it.

Reply:
Sure, they just ripped loads out of Email Alert API so maybe can find them in the past, though it'd be best to have a current one so we can look at it in icinga.

Very confusing 🤔

Reply:
ah found it https://github.com/alphagov/email-alert-api/commit/0c17e757179cb4b0308bf0b49bb3b19842c15bc2#diff-9fccbed8f77bf59ae535edc077ce5e21

Reply:
doesn't mean that it worked but I'm glad I wasn't losing my marbles. I might test it in integration this afternoon, as it stands visiting `/healthcheck` will show both results but I don't know what it'll do in icinga so I'll be able to see what actually works.

Reply:
Ah I see, hmm that's a bit weird too as the details for that would just be numbers if I'm understanding that correctly rather than strings 🤔

Reply:
It's all weird! I'm trying to test this in integration, I've managed to make it give a warning but not actually raise anything in icinga. This is because of an outstanding puppet change so I'll be looking to deploy puppet to integration in order to test this. I just want to make sure that the messages show up how I expect in Icinga. I'm assuming that we don't really care what it looks like when a person visits `/healthcheck`?

Reply:
This was resolved by checking in integration. Turns out that the details method doesn't seem to be shown anywhere. I've replaced it with the "message" method that will show up in Icinga. I've also made changes to the doc to make it clearer what's happening here https://github.com/alphagov/govuk_app_config/pull/135